site/docs/reference/Connectors/materialization-connectors/Rockset.md

```diff
@@ -42,4 +45,84 @@ materializations:
       	workspace: ${namespace_name}
       	collection: ${table_name}
     source: ${tenant}/${source_collection}
-```
\ No newline at end of file
+```
+## Bulk ingestion for large backfills of historical data
+
+You can backfill large amounts of historical data into Rockset using a *bulk ingestion*. Bulk ingestion must originate in S3 and requires additional steps in your dataflow. Flow's Rockset connector supports this through the GitOps workflow.
+
+### Prerequisites
+
+Before completing this workflow, make sure you have:
+* A working catalog spec including at least one Flow collection with its schema specified.
+* A production or [development](../../../getting-started/installation.md) environment
+::: tip
+The following is an intermediate workflow. As needed, refer to [this guide](../../../guides/create-dataflow.md) for the basic steps to create and deploy a catalog spec using the GitOps workflow.
+:::
+
+### How to configure a bulk ingestion
+
+A bulk ingestion from a Flow collection into Rockset is essentially a two-step process. First, Flow writes your historical data into an S3 bucket using Estuary's [materialize-s3-parquet](../materialize-s3-parquet/) connector. Once the data is caught up, it uses the Rockset connector to backfill the data from S3 into Rockset and then switch to the Rockset Write API for the continuous materialization of new data.
+
+import Mermaid from '@theme/Mermaid';
```

Comment:
I'm not positive, but I think this `import` might need to be at the top of the file. Or at least it probably _should_ be at the top to be consistent with prevailing conventions.

Reply:
It does work in the test output, and I based this off of how Johnny had set up his diagrams in some of the docs he created... that is a good point about the convention but I think I'll leave it for now because of those other files

Reply:
It does work in the test output, and I based this off of how Johnny had set up his diagrams in some of the docs he created... that is a good point about the convention but I think I'll leave it for now because of those other files